target/linux/ramips/image/mt7621.mk

```diff
@@ -1685,6 +1685,31 @@ define Device/zio_freezio
 endef
 TARGET_DEVICES += zio_freezio
 
+define Device/zyxel_keenetic-giga-iii
+  $(Device/dsa-migration)
+  KERNEL_SIZE := 4096k  # kernel partition size
+  IMAGE_SIZE := 50m
+  BLOCKSIZE := 128k     # NAND erase size (block size)
+  PAGESIZE := 2048      # NAND page size
+  DEVICE_VENDOR := ZyXEL
+  DEVICE_MODEL := Keenetic Giga III
+  SUPPORTED_DEVICES += kng_re
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb3
+  DEVICE_PACKAGES += kmod-usb-ledtrig-usbport kmod-usb-storage
+  VER_NUM := $$(shell cut -d '-' -f 1 <<< "$$(VERSION_NUMBER)" )
+  VER_DATE := $$(shell date --utc -d @$$(SOURCE_DATE_EPOCH) +%y%m%d)
+  VER_FULL := $$(VER_NUM) $$(VER_DATE)
+  KERNEL := kernel-bin | append-dtb | lzma | loader-kernel | uImage none -n 'KN-GIGA3 $$(VERSION_DIST) $$(VER_FULL)'
```

Comment:
please create a Build recipe for that instead of introducing local variables here.

Reply:
I think this is completely superfluous.
Did you talk about something similar?
```
define Device/repo-version
  REPO_VER := $$(shell cut -d '-' -f 1 <<< "$$(VERSION_NUMBER)" )
  REPO_DATE := $$(shell date --utc -d @$$(SOURCE_DATE_EPOCH) +%y%m%d)
  REPO_VER_DATE := $$(REPO_VER) $$(REPO_DATE)
  REPO_FULL_VER := $$(VERSION_DIST) $$(REPO_VER) $$(REPO_DATE)
endif
```

Reply:
No, this is a common device definition. I'm talking about a build recipe, these are the blocks starting with "Build/", i.e. those you are using when you put all these nice terms like "kernel-bin" or "loader-kernel" ...

Reply:
Some thing like this (just typed it without too much care):

```
define Build/my_special_uimage_command
  $(eval repo_ver=$$(shell cut -d '-' -f 1 <<< "$$(VERSION_NUMBER)"))
  $(eval repo_date=$$(shell date --utc -d @$$(SOURCE_DATE_EPOCH) +%y%m%d))
  uImage none -n '$(word 1, $(1)) $$(VERSION_DIST) $(repo_ver) $(repo_date)'
endef

[...]
KERNEL := kernel-bin | append-dtb | lzma | loader-kernel | my_special_uimage_command KN-GIGA3
[...]
```

Reply:
Maybe you can then do what you like with a separate commit?
I really think that making a specially separate recipe for one device is a complete overkill.

Reply:
And in which file should this recipe be placed?
Is it possible in `target/linux/ramips/image/Makefile` ?

Reply:
This depends on the devices it will be used for. If it's just for one device, you can put in the same file as the device definition (of course, at the top and keeping proper sorting of recipes).